command/cp.go

```diff
@@ -537,18 +539,20 @@ func (c Copy) doDownload(ctx context.Context, srcurl *url.URL, dsturl *url.URL)
 	if err != nil {
 		// file must be closed before deletion
 		file.Close()
-		dErr := dstClient.Delete(ctx, dsturl)
+		dErr := dstClient.Delete(ctx, &url.URL{Path: file.Name(), Type: dsturl.Type})
 		if dErr != nil {
 			printDebug(c.op, dErr, srcurl, dsturl)
 		}
 		return err
 	}
-	defer file.Close()
 
 	if c.deleteSource {
 		_ = srcClient.Delete(ctx, srcurl)
 	}
 
+	file.Close()
+	dstClient.Rename(file, dsturl.Absolute())
```

Comment:
Please handle error case

Reply:
Do we really need to handle file closing errors? The rest of the code doesn't handle errors for `file.Close()`. If I do, it will look like this:

```code
size, err := srcClient.Get(ctx, srcurl, file, c.concurrency, c.partSize)
err2 := file.Close()
if err2 != nil {
	return err2
}
if err != nil {
	dErr := dstClient.Delete(ctx, &url.URL{Path: file.Name(), Type: dsturl.Type})
	if dErr != nil {
		printDebug(c.op, dErr, srcurl, dsturl)
	}
	return err
}
```

Reply:
Not need to handle closing errors better to handle `Rename()`.